File: frontend/src/types/document.ts
export interface DocumentItem {
  id: string;
  session_id?: string;
  name: string;
  mime_type: string;
  version: number;
  checksum?: string;
  storage_key?: string;
  status: "uploaded" | "parsing" | "parsed" | "indexing" | "ready" | "failed";
  created_by?: string;
  created_at?: string;
}

export interface CCOClaim {
  id: string;
  text: string;
  source_sentence?: string;
  confidence: number;
  evidence_refs?: string[];
  status?: string;
}

export interface CCOData {
  document_id: string;
  cco_version_id: string;
  version: number;
  hash: string;
  title?: string;
  executive_overview?: string;
  claims: CCOClaim[];
  identifiers?: string[];
  key_findings?: string[];
  cco_json?: Record<string, any>;
}

export interface EvidenceChunk {
  chunk_id: string;
  text: string;
  section: string;
  page: number;
}

File: backend/app/schemas/document.py
"""
ContentForge AI — Document Schemas
"""

from datetime import datetime
from typing import Any, Optional
from pydantic import BaseModel, Field
from app.schemas.enums import DocumentStatus


class DocumentResponse(BaseModel):
    id: str
    session_id: Optional[str] = None
    name: str
    mime_type: str
    version: int = 1
    checksum: Optional[str] = None
    storage_key: Optional[str] = None
    status: DocumentStatus = DocumentStatus.UPLOADED
    created_by: Optional[str] = None
    created_at: Optional[datetime] = None


class DocumentVersionResponse(BaseModel):
    id: str
    document_id: str
    version: int
    checksum: Optional[str] = None
    storage_key: Optional[str] = None
    created_at: Optional[datetime] = None


class DocumentCCOResponse(BaseModel):
    document_id: str
    cco_version_id: str
    version: int
    hash: str
    source_block_count: int = 0
    claim_count: int = 0
    cco_json: dict[str, Any] = {}


class DocumentEvidenceResponse(BaseModel):
    document_id: str
    chunk_count: int = 0
    chunks: list[dict[str, Any]] = []
    source_blocks: list[dict[str, Any]] = []
